### DISCUSSION TITLE
Launching a RUN_COMMAND Android Intent from Unity app?

### DISCUSSION THREAD

**Discussion by saricden** [termux/termux-app#2867] [Q&A] [answered]
Hi everyone,

I'm new to Android development, Intents, and Unity in general. That said, I am required to use Unity to build my Android app due to a specific SDK requirement that basically powers my entire app.

I am looking for a way to effectively emulate a Termux session inside my own terminal UI within my 3D Unity scene. I already have the UI setup, and have followed the instructions for setting up the RUN_COMMAND Intent [described here](https://github.com/termux/termux-app/wiki/RUN_COMMAND-Intent).

What I am looking for is an example of how one could issue this Intent with the extra "com.termux.RUN_COMMAND_BACKGROUND" set to true.

**As a super basic example**, say I wanted to start a Termux session and simply send the command `ls` then `pwd`.

Where would I start?

**Comment by SunPodder** [comment, marked as answer]
The docs clearly have a example [here](https://github.com/termux/termux-app/wiki/RUN_COMMAND-Intent#BasicExamples) 

However here is what you need
```java
Intent intent = new Intent();
intent.putExtra("com.termux.RUN_COMMAND_BACKGROUND", true);
startService(intent);
```
